libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -37,6 +37,24 @@ lua_scripts::lua_scripts(const AP_Int32 &vm_steps, const AP_Int32 &heap_size, co
     _heap = hal.util->allocate_heap_memory(heap_size);
 }
 
+lua_scripts::~lua_scripts() {
+    repl_cleanup();
```

Comment:
It's unclear to me that we need this. It's only called at the end of doREPL(), and that's the only time we have any REPL running meaningfully.

If there was a session open calling it here actually has a race condition and will leak extra file handles. (We start cleaning up the session, then get preempted after setting `terminal.session` to false and a MAVLink packet is pending, which will then leak the file handles that we had before). Arguably when we make a new REPL we should just clean the temp files then.

Reply:
removed, I was just collecting everything that looked like it may need to be cleaned up..